src/tunnel_controller.rs

```diff
@@ -89,3 +96,52 @@ impl Tunnel {
         }
     }
 }
+
+
+#[cfg(test)]
+describe! tunnel {
+
+    before_each {
+        use std::thread::sleep;
+        use std::time::Duration;
+        use stubs::tunnel::TunnelConfigStub;
+
+        let config = TunnelConfigStub::new(1, "a".to_owned());
+        let mut tunnel = super::Tunnel::new(config.clone());
+
+        tunnel.start().unwrap();
+        // XXX Sleep needed otherwise the process is not started in time
```

Comment:
`should_start_a_new_tunnel` and `should_stop_a_tunnel` fail. The count has a value decreased by one (compared to the expected value).